### PR TITLE
chore: convert functions to an ES2015 classes

### PR DESCRIPTION
**Pull request by hasanalkaf3**
[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

- [ ] Add an algorithm?
- [ ✅ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### Checklist:

- [ ✅ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [ ✅ ] This pull request is all my own work -- I have not plagiarized.
- [ ✅ ] I know that pull requests will not be merged if they fail the automated tests.
- [ ] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [ ✅ ] All new JavaScript files are placed inside an existing directory.
- [ ✅ ] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [ ✅ ] All new algorithms have a URL in their comments that points to Wikipedia or another similar explanation.
- [ ✅ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: Data-Structures/Array/Reverse.js [hunk 1]
```diff
@@ -7,11 +7,9 @@
 
 const Reverse = (arr) => {
   // limit specifies the amount of Reverse actions
-  for (let i = 0, j = arr.length - 1; i < arr.length / 2; i++, j--) {
-    const temp = arr[i]
-    arr[i] = arr[j]
-    arr[j] = temp
-  }
+  for (let i = 0, j = arr.length - 1; i < arr.length / 2; i++, j--)
+    [arr[i], arr[j]] = [arr[j], arr[i]] // Swapping elements ES6 way
```
Comment (appgurueu) [inline Data-Structures/Array/Reverse.js:11]: Unnecessary comment.

File: Data-Structures/Stack/Stack.js [hunk 2]
```diff
@@ -9,45 +9,47 @@
...
-    // The array representation of the stack
-    this.stack = []
-  }
-
-  // Adds a value onto the end of the stack
-  Stack.prototype.push = function (value) {
-    this.stack[this.top] = value
-    this.top++
-  }
+  class Stack {
```
Comment (appgurueu) [inline Data-Structures/Stack/Stack.js:12]: This should be just one `Stack` class. There is no reason for this to be within a function as far as I can see.

File: Data-Structures/Tree/AVLTree.js [hunk 3]
```diff
@@ -32,23 +32,62 @@ let utils
  * If no argument is sent it uses utils.comparator
  */
 const AVLTree = (function () {
-  function _avl(comp) {
-    /** @public comparator function */
-    this._comp = undefined
-    this._comp = comp !== undefined ? comp : utils.comparator()
+  class _avl {
```
Comment (appgurueu) [inline Data-Structures/Tree/AVLTree.js:35]: Same here. These classes should be toplevel.

**Review by appgurueu — COMMENTED** [review]
Thanks for the refactor. In general, please try to remove unnecessary functions as well. Classes that are unnecessarily "nested" inside functions are confusing.

**Review by appgurueu — APPROVED** [review]
Decent improvement as-is.

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**